encrypt.go

```diff
@@ -0,0 +1,77 @@
+package dht
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"crypto/sha256"
+	"errors"
+
+	"github.com/multiformats/go-multihash"
+)
+
+const (
```

Comment:
Would it be possible to get these constants from somewhere where they are already defined instead of defining them again? This would help with code maintainability :)

Reply:
unfortunately the Go AES package doesn't define these :/ I'll try looking somewhere for them!

Reply:
We should be able to get Nonce size from [`cipher.AEAD`](https://pkg.go.dev/crypto/cipher#AEAD.NonceSize) interface.

Reply:
updated